### PR TITLE
LBAC-32 Added Tests for LocationUtils

### PR DESCRIPTION
**Pull request by VANKINEENITAWRUN**
## Issue I worked on

see https://issues.openmrs.org/browse/LBAC-32

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

 
- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by suthagar23** [comment]
Can you please follow the guidelines as per https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by VANKINEENITAWRUN** [comment]
can you mention which thing I forgot, since I can't find anything wrong with PR as per PR guidelines?

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+		locationService = Context.getLocationService();
+		executeDataSet(XML_FILENAME_WITH_PERSON_DATA);
+	}
+
+	@Test
+	public void getPersonLocation_shouldReturnPersonLocation(){
+		Person person3 = personService.getPerson(DEMO_PERSON3_ID);
+		Location location = locationService.getLocation(DEMO_LOCATION1_ID);
+		Location location3 = LocationUtils.getPersonLocation(person3);
+		assertEquals(location,location3);
```
Comment (suthagar23) [inline api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java]: How can you ensure that, if the LocationUtils.getPersonLocation() returns the same location as locationService.getLocation()?

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+		Person person2 = personService.getPerson(DEMO_PERSON2_ID);
+		Location location2 = LocationUtils.getPersonLocation(person2);
+		assertNull(location2);
```
Comment (suthagar23) [inline api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java]: This should go under a separate test case like ```shoudNot...```

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+		Person person4 = personService.getPerson(DEMO_PERSON4_ID);
+		Location location4 = LocationUtils.getPersonLocation(person4);
+		assertNotEquals(location3,location4);
```
Comment (suthagar23) [inline api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java]: This also should go under a separate test case like ```shoudNot...```

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java (hunk 1 above)
Comment (VANKINEENITAWRUN) [reply api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java]: Here in the Dataset person with Id `DEMO_PERSON3_ID` is associated with the location having an ID  `DEMO_LOCATION1_ID`.

Comment (suthagar23) [reply api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java]: > Dataset person with Id DEMO_PERSON3_ID is associated with the location having an ID DEMO_LOCATION1_ID

Here, How can you show it through the code? 
For a unit test, we should not attach any human justifications for the results. Here you trust your Dataset and assuming that the person has that one.

A good way would be, 
1. Create a location - LocationA
2. Create a person
3. Add location to that person
4. Get the location from the person through our method. 
5. Now check for the response against LocationA
All of the above methods should be implemented through the codes.

Comment (suthagar23) [reply api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java]: Follow it for other test cases as well !!, You can get anything from Dataset, but you need to add test to ensure that you are getting same from the dataset

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+	@Test
+	public void getPersonLocation_shouldReturnNullIfNoLocationAttribute() {
+		Person person2 = personService.getPerson(DEMO_PERSON2_ID);
+		Location location = LocationUtils.getPersonLocation(person2);
+		assertNull(location);
+	}
+
+	@Test
+	public void doesPersonBelongToGivenLocation_True_ifPersonBelongToSameLocation(){
```
Comment (suthagar23) [inline api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java]: Can you change this method name to a meaning full? 
Eg : does<A>ShouldBeTrue_If<B>

**Comment by suthagar23** [comment]
@VANKINEENITAWRUN Once you made changes on the PR, please notify me through the talk forum or message, Otherwise we may miss it

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+
+	@Test
+	public void getPersonLocation_shouldReturnPersonLocation(){
+		Person person3 = personService.getPerson(DEMO_PERSON3_ID);
+		Location location1 = locationService.getLocation(DEMO_LOCATION1_ID);
+		PersonAttributeType personAttributeType = personService.getPersonAttributeType(DEMO_PERSONATTRIBUTE1_ID);
+		PersonAttribute personAttribute = new PersonAttribute(personAttributeType,location1.getUuid());
+		person3.addAttribute(personAttribute);
+		Location location3 = LocationUtils.getPersonLocation(person3);
+		assertEquals(location3,location1);
```
Comment (suthagar23) [inline api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java]: Did you confirm that the person doesn't have this person Attribute or Location before you adding the person attribute?

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+		assertNotEquals(location1.getUuid(),location2.getUuid());
+		PersonAttribute personAttribute = new PersonAttribute(personAttributeType,location1.getUuid());
+		person2.addAttribute(personAttribute);
+		String sessionLocationUuid2 = location2.getUuid();
+		Boolean aBoolean2 = LocationUtils.doesPersonBelongToGivenLocation(person2,personAttributeType,sessionLocationUuid2);
+		assertFalse(aBoolean2);
+	}
+
+	@Test
+	public void doesUserBelongToGivenLocationShouldBeTrue_ifUserBelongToSameLocation(){
```
Comment (suthagar23) [inline api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java]: it shold be changed as ```doesUserBelongToGivenLocation_shouldBeTrueIfUserBelongToSameLocation```? For other method as well

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+		assertEquals(0,user1.getUserProperties().size());
+		Location location1 = locationService.getLocation(DEMO_LOCATION1_ID);
+		user1.setUserProperty(LocationBasedAccessConstants.LOCATION_USER_PROPERTY_NAME,location1.getUuid());
+		String sessionLocationUuid = location1.getUuid();
+		Boolean aBoolean  = LocationUtils.doesUserBelongToGivenLocation(user1,sessionLocationUuid);
+		assertTrue(aBoolean);
+	}
+
+	@Test
+	public void doesUserBelongToGivenLocationShouldBeFalse_ifUserNotBelongToSameLocation() {
```
Comment (suthagar23) [inline api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java]: Have you formatted the code? Check other places as well

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java (hunk 6 above)
Comment (VANKINEENITAWRUN) [reply api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java]: @suthagar23  fixed can check PR now.

File: api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java (hunk 5 above)
Comment (VANKINEENITAWRUN) [reply api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java]: Added `AssertNull` for checking attributes. Should be fine now.

**Pull request merged**